src/Comms.java

```diff
@@ -0,0 +1,31 @@
+import java.util.Scanner;
+public class Comms {
```

Comment:
i like this - separating the terminal IO from the file IO is probably even better than the way I did it. Nice work.